frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java

```diff
@@ -0,0 +1,140 @@
+package benchmark;
+
+import io.micronaut.context.annotation.Property;
+import io.micronaut.http.context.ServerRequestContext;
+import io.micronaut.http.server.netty.NettyHttpRequest;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFactory;
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.socket.DatagramChannel;
+import io.netty.channel.socket.InternetProtocolFamily;
+import io.netty.util.concurrent.FastThreadLocal;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.datagram.DatagramSocketOptions;
+import io.vertx.core.impl.VertxBuilder;
+import io.vertx.core.net.ClientOptionsBase;
+import io.vertx.core.net.NetServerOptions;
+import io.vertx.core.spi.transport.Transport;
+import io.vertx.pgclient.PgConnectOptions;
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.PoolOptions;
+import jakarta.inject.Singleton;
+
+import java.net.SocketAddress;
+import java.util.concurrent.ThreadFactory;
+
+@Singleton
+public class ConnectionHolder {
+    private final FastThreadLocal<Future<PgConnection>> conn = new FastThreadLocal<>();
+
+    @Property(name = "datasources.default.url") String url;
+    @Property(name = "datasources.default.username") String user;
+    @Property(name = "datasources.default.password") String password;
+    @Property(name = "datasources.default.maximum-pool-size") int maxPoolSize;
+
+    public Future<PgConnection> get() {
+        Future<PgConnection> c = conn.get();
+        if (c == null) {
+            NettyHttpRequest<?> req = (NettyHttpRequest<?>) ServerRequestContext.currentRequest().get();
```

Comment:
i think i prefer ThreadExecutorMap, unless you have a reason to use the ServerRequestContext

Reply:
but the request is only used to get the event loop.

Reply:
just cast it. i think it should work

Reply:
Ok

Reply:
thanks. does it work? :sweat_smile: